tools/2024-06-gitops-feature-spec.md

```diff
@@ -0,0 +1,374 @@
+Our goal is to deliver a generalized model such that Flux, ArgoCD plus future GitOps technologies can all be integrated in a consistent manner.  How feasible is that?  We'll need to learn more about the GitOps tool landscape to determine feasibility.
+
+We as the Radius team understand that while Flux and ArgoCD might currently be the most popular options utilized by customers and while the current scenarios might reference Flux or ArgoCD, the solution we will build will be abstract enough to support all other high priority GitOps toolsets. We currently understand the GitOps workflow if abstracted in implementation usually follows this format:
+1. Install GitOps tool onto Kubernetes cluster. 
+2. Add git repository information to GitOps tool
+3. Deploy application to Kubernetes cluster 
+4. Done 
+ 
+> The implementation settled on is generic enough to support GitOps on a wide level not narrowed down to just Flux or ArgoCD.
+
+## Current state
+N/A
+
+# Scenario 1: Install and configure Radius using GitOps
+
+## Target customers
+<!-- Of the customers / personas listed in the Epic doc, what subset are we delivering this scenario to serve? -->
+**Infrastructure operators / administrators** - In enterprise applications teams that use GitOps, operators are responsible for configuring and managing the git repos for infrastructure and application configuration. They configure GitOps policies as well as the Kubernetes manifests, Helm charts, etc. that are applied to the application and the Kubernetes cluster.
+
+## Details of customer problem
+<!-- <Write this in first person. You basically want to summarize what “I” as a customer am trying to accomplish, why the current experience is a problem and the impact it has on me, my team, my work and or biz, etc…. i.e. “When I try to do x aspect of cloud native app development, I have the following challenges / issues….<details>. Those issues result in <negative impact those challenges / issues have on your work and or business.> -->
+As an infrastructure operator in an enterprise applications team that has to maintain complex Kubernetes clusters and the platforms that run on them, I need to ensure that my team can deploy and manage applications and infrastructure in a consistent and reliable way. Thus, we make use of GitOps to manage Kubernetes clusters and applications that run in those clusters. All relevant configuration information is stored in git repositories that serve as the source of truth for both the application code and infrastructure configuration. Most commonly, infrastructure configuration and application code are housed in separate repos. To use Radius, I need to include Radius as a dependency for my Kubernetes cluster, just as I would any other dependency, via my GitOps tools. Without clear guidance on how to do this, I am unsure how to use both GitOps and Radius for a better together experience.
```

Comment:
Instead of making it a dependency, can't we write code that detects if Radius is installed or not, and if not, GitOps can install Radius?

Reply:
Specifying the dependency (instead of auto-installing) is the desirable pattern for GitOps as operators would want explicit control over things like versioning.